[FEATURE]: Dont bundle the translations in the manager-react-components

helabenkhalfallah opened the pull request:
ref: MANAGER-18402

<!--
Hello 👋 Thank you for submitting a Pull Request.

Have any questions? Check out the contributing docs at https://github.com/ovh/manager/blob/master/CONTRIBUTING.md

Please read the following before submitting:

- Keep your PR as small as possible to make easy reviews.
- Commits are signed-off.
- Update only Messages_fr_FR.json locales for any content changes.
- Lint has passed locally.
- Standalone app was ran and tested locally.
- Ticket reference is mentioned in linked commits (internal only).
- Breaking change is mentioned in relevant commits.
- Limit your PR to one type (build, ci, docs, feat, fix, perf, refactor, revert, style, test or release)
-->

## Description

Original issue: https://github.com/ovh/manager/issues/13901

## Additional Information

Before: 
<img width="1470" alt="BEFORE_1" src="https://github.com/user-attachments/assets/5ddae8fd-2dba-4beb-bf49-ad59eb2c7d1e" />

After:
<img width="1470" alt="AFTER_1" src="https://github.com/user-attachments/assets/e2a2702b-7b46-4e61-94a3-6a06eeaa221e" />

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_de_DE.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_en_GB.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_es_ES.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_fr_FR.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_it_IT.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_pl_PL.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

helabenkhalfallah replied:
Yes I'll clean all :)

anooparveti commented on packages/manager-react-components/src/components/ManagerText/translations/Messages_pt_PT.json:
Are you sure to merge this change in `develop`. I assume this change is for testing purpose.

anooparveti commented on packages/manager-react-components/src/utils/TranslationUtils.ts:
```diff
@@ -0,0 +1,85 @@
+
+  i18next.on('languageChanged', handleLanguageChange);
+
+  window.addEventListener('languagechange', () => {
+    const normalizedLang = normalizeLanguageCode(navigator.language);
+    i18next.changeLanguage(normalizedLang);
+  });
```
**nitpick(performance):** Since this function needs to be called for every component that uses translations, we have an added eventListener. So imo, it's not good to have 1 eventListener per component and moreover I dont see that it's being removed when the component is removed from the DOM.

anooparveti reviewed (changes requested):
<img width="1439" alt="Screenshot 2025-05-29 at 12 54 22 PM" src="https://github.com/user-attachments/assets/7711cca6-9840-4d76-8b65-291189bc313e" />

Irrespective of whether a component is present in the page or not, existing translations of all the components are loaded by default. **IMO, it's bad and not scalable when we have lots of components with translations, difference can be significant to load all the unwanted translations.**

I think there will be an issue with fallback language as well. Current behavior, if translation key does not exist on the current locale file, content from fallback language (FR) will be displayed. **Since the current implementation does not load/configure the fallback language, there might be an issue with the current expected behaviour.**

helabenkhalfallah commented:
> <img alt="Screenshot 2025-05-29 at 12 54 22 PM" width="1439" src="https://private-user-images.githubusercontent.com/22835814/448730246-7711cca6-9840-4d76-8b65-291189bc313e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg1ODc3MzYsIm5iZiI6MTc0ODU4NzQzNiwicGF0aCI6Ii8yMjgzNTgxNC80NDg3MzAyNDYtNzcxMWNjYTYtOTg0MC00ZDc2LThiNjUtMjkxMTg5YmMzMTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTMwVDA2NDM1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5NWI3OGVhNDc0MWFiNjUyZGNkYzg0NWY0MWU3NDc1M2RhZjgzMjgyY2MwYjZmMzFlYjc4MjliZWJjYjVkZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1hmd9n4ji51gjA3-CsLyGzfyROktyXuZnjIiBCn8_ss">
> Irrespective of whether a component is present in the page or not, existing translations of all the components are loaded by default. **IMO, it's bad and not scalable when we have lots of components with translations, difference can be significant to load all the unwanted translations.**
> 
> I think there will be an issue with fallback language as well. Current behavior, if translation key does not exist on the current locale file, content from fallback language (FR) will be displayed. **Since the current implementation does not load/configure the fallback language, there might be an issue with the current expected behaviour.**

Did you see that I did not change anything comparing to what's loaded before? I just changed  the type of import The same code is preserved and I moved it to a new utils file. So what behaviour did I change? 

> Since the current implementation does not load/configure the fallback language, there might be an issue with the current expected behaviour.

I did not change on any application file, only in MRC:

<img width="1115" alt="Capture d’écran 2025-05-30 à 08 42 49" src="https://github.com/user-attachments/assets/548d3240-a341-4b6b-95a0-e0c20adfc721" />
<img width="1115" alt="Capture d’écran 2025-05-30 à 08 43 14" src="https://github.com/user-attachments/assets/91c76920-fd4d-43c4-a924-09884ae0bb13" />

"Since the current implementation does not load/configure the fallback language, there might be an issue with the current expected behaviour."

helabenkhalfallah commented:
> <img alt="Screenshot 2025-05-29 at 12 54 22 PM" width="1439" src="https://private-user-images.githubusercontent.com/22835814/448730246-7711cca6-9840-4d76-8b65-291189bc313e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg1ODc3MzYsIm5iZiI6MTc0ODU4NzQzNiwicGF0aCI6Ii8yMjgzNTgxNC80NDg3MzAyNDYtNzcxMWNjYTYtOTg0MC00ZDc2LThiNjUtMjkxMTg5YmMzMTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTMwVDA2NDM1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5NWI3OGVhNDc0MWFiNjUyZGNkYzg0NWY0MWU3NDc1M2RhZjgzMjgyY2MwYjZmMzFlYjc4MjliZWJjYjVkZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1hmd9n4ji51gjA3-CsLyGzfyROktyXuZnjIiBCn8_ss">
> Irrespective of whether a component is present in the page or not, existing translations of all the components are loaded by default. **IMO, it's bad and not scalable when we have lots of components with translations, difference can be significant to load all the unwanted translations.**
> 
> I think there will be an issue with fallback language as well. Current behavior, if translation key does not exist on the current locale file, content from fallback language (FR) will be displayed. **Since the current implementation does not load/configure the fallback language, there might be an issue with the current expected behaviour.**

As we see together:
-> I'll take a look at the fallback language.
-> But for the chuncks that are all loaded, this is another concern related to MRC. MRC should be modular and tree-shaked so language chuncks can be loaded as need. This is a concern of another ticket.

Thanks :)

helabenkhalfallah commented:
## Handling default fallback

**The `test-key` is not declared for `pl_PL`, `it_IT` and `pt_PT`..**

### Before

We can see that the default French language is not loaded:
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/a15dad02-14e2-418a-bf18-69210172b8d3" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/c7b52159-1397-4cb0-9a0a-037e4770aa46" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/76d756ae-50a5-4a4e-a738-5f8a43bfd2e5" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/4d6ca7af-6155-47b3-8cdc-4e62194b6ef4" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/46ffb55e-df48-425a-861f-c18294331bc4" />

## After

We can see below that the French language is used as fallback: 
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/77d3ca39-6797-434d-b910-41653261876e" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/d824e8d6-e620-4b0d-854c-fa205318759f" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/9f574e66-99e2-4691-bc1b-2cb5042e572d" />
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/6efa5e3b-0d84-4837-92d7-293a473a9c12" />

anooparveti reviewed (dismissed):
For the namespace, I think it's better to add prefix like `@ovh-ux/manager-react-components` to avoid collision if the µ-app also add a same namespace.
For example, in `iam` µ-app, we can have a namespace with name `iam` and we also have a namespace `iam` in MRC.
